### PR TITLE
Feature/obstacle minigame

### PR DESCRIPTION
**Pull request by Foref**
Contains new screens for obstacle-minigame (SpaceMapScreen) and extractor-minigame (ExtractorMiniGameScreen). Access new screens through respective main menu button.

Inside SpaceMapScreen contains SpaceGameArea with Spaceship and Asteroid. (And a background music)
Nothing inside ExtractorMiniGameScreen

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java [hunk 1]
```diff
@@ -104,8 +108,17 @@ public void create() {
     spawnTrees();
     spawnPowerups();
     spawnExtractors();
+
+
+    var player = spawnPlayer();
```
Comment (rowan-gray) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java]: Are there meant to be 3 spawnPlayers?

File: .DS_Store
Comment (rowan-gray) [inline .DS_Store:1]: Add a git ignore for this.

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java
Comment (rowan-gray) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:1]: Make sure you're only spawning one player.

**Review by rowan-gray — CHANGES_REQUESTED** [review]
Mostly looks good. Just double check whether there is meant to be 3 players spawned in ForestGameArea and add a git ignore for .DS_Store

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java (hunk 1 above)
Comment (Foref) [reply source/core/src/main/com/csse3200/game/areas/ForestGameArea.java]: nope, missed that

File: .DS_Store
Comment (Foref) [reply .DS_Store:1]: ok

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java
Comment (Foref) [reply source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:1]: done

Comment (rowan-gray) [reply source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:1]: Thanks!

**Review by rowan-gray — APPROVED** [review]
All good 😄

**Pull request merged**